harness/determined/tensorboard/base.py

```diff
@@ -95,3 +97,30 @@ def get_metric_writer() -> tensorboard.BatchMetricWriter:
         writer = pytorch.TorchWriter()
 
     return tensorboard.BatchMetricWriter(writer)
+
+
+class _TensorboardUploadThread(threading.Thread):
+    def __init__(
+        self, upload_function: Callable[[List[pathlib.Path]], None], work_queue: queue.Queue
```

Comment:
What do you think about hiding the queue inside the upload thread, rather than passing it in?

So you'd use it like this:

```python
upload_thread = _TensorboardUploadThread(my_upload)
upload_thread.start()
upload_thread.upload(my_paths)  # this happens asynchronously
```

Reply:
Great point Ryan, I have actually thought about hiding the queue inside of the upload thread and I have some questions about performance implication of this approach: 

If the upload thread is in the process of uploading, then when main thread call upload_thread.upload(my_paths), would that call block until upload thread is done uploading?

If so, it may be better to not hide the queue from the main thread.

Reply:
No, `_TensorboardUploadThread.upload()` would be defined like this:

```python
def upload(self, paths):
    self._work_queue.put(paths)
```

Reply:
oh, also I'd recommend adding a `_TensorboardUploadThread.close()` method that actually sends the None sentinel, then calls `self.join()` to ensure the thread is actually done uploading.

Reply:
Sounds great! Will do so!